lib/active_merchant/billing/gateways/flex_charge.rb

```diff
@@ -0,0 +1,244 @@
+          gsub(%r(("cardNumber\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]').
+          gsub(%r(("verification_value\\?":\\?")\d+), '\1[FILTERED]')
+      end
+
+      private
+
+      def add_merchant_data(post, options)
+        post[:siteId] = @options[:site_id]
+        post[:mid] = @options[:mid]
+      end
+
+      def add_base_data(post, options)
+        post[:isDeclined] = cast_bool(options[:is_declined])
+        post[:orderId] = options[:order_id]
+        post[:idempotencyKey] = options[:idempotency_key] || options[:order_id]
+      end
+
+      def add_mit_data(post, options)
+        return unless options[:is_mit].present?
+
+        post[:isMIT] = cast_bool(options[:is_mit])
+        post[:isRecurring] = cast_bool(options[:is_recurring])
+        post[:expiryDateUtc] = options[:mit_expiry_date_utc]
+      end
+
+      def add_customer_data(post, options)
+        post[:payer] = { email: options[:email] || 'NA', phone: phone_from(options) }.compact
+      end
+
+      def add_address(post, payment, address)
+        first_name, last_name = address_names(address[:name], payment)
+
+        post[:billingInformation] = {
+          firstName: first_name,
+          lastName: last_name,
+          country: address[:country],
+          phone: address[:phone],
+          countryCode: address[:country],
+          addressLine1: address[:address1],
+          state: address[:state],
+          city: address[:city],
+          zipCode: address[:zip]
+        }.compact
+      end
+
+      def add_invoice(post, money, credit_card, options)
+        post[:transaction] = {
+          id: options[:order_id],
+          dynamicDescriptor: options[:description],
+          timestamp: Time.now.utc.iso8601,
+          timezoneUtcOffset: options[:timezone_utc_offset],
+          amount: money,
+          currency: (options[:currency] || currency(money)),
+          responseCode: options[:response_code],
+          responseCodeSource: options[:response_code_source] || '',
+          avsResultCode: options[:avs_result_code],
+          cvvResultCode: options[:cvv_result_code],
+          cavvResultCode: options[:cavv_result_code],
+          cardNotPresent: credit_card.verification_value.blank?
+        }.compact
+      end
+
+      def add_payment_method(post, credit_card, address, options)
+        post[:paymentMethod] = {
+          holderName: credit_card.name,
+          cardType: 'CREDIT',
+          cardBrand: credit_card.brand&.upcase,
+          cardCountry: address[:country],
+          expirationMonth: credit_card.month,
+          expirationYear: credit_card.year,
+          cardBinNumber: credit_card.number[0..5],
+          cardLast4Digits: credit_card.number[-4..-1],
+          cardNumber: credit_card.number,
+          Token: false
+        }.compact
+      end
+
+      def address_names(address_name, payment_method)
+        split_names(address_name).tap do |names|
+          names[0] = payment_method&.first_name unless names[0].present?
```

Comment:
this can be simplified with something like this no?

```ruby
split_names(address_name).tap do |names|
  names[0] ||= payment_method&.first_name
  names[1] ||= payment_method&.last_name
end
```

Reply:
Yup, that was my initial intention, but sadly with just on name for example "Doe", what `split_names`gives you is a ["", "Doe"]